eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/immutable/ImmutableBagFactoryImpl.java

```diff
@@ -105,4 +108,55 @@
         }
         return this.of((T[]) Iterate.toArray(items));
     }
+
+    @Override
+    public <T> ImmutableBag<T> withOccurrences(T element, int occurrence)
+    {
+        MutableBag<T> bag = new HashBag<>(occurrence);
+        bag.addOccurrences(element, occurrence);
+
+        return bag.toImmutable();
+    }
+
+    @Override
+    public <T> ImmutableBag<T> withOccurrences(T element1, int occurrence1, T element2, int occurrence2)
+    {
+        MutableBag<T> bag = new HashBag<>(occurrence1 + occurrence2);
+        bag.addOccurrences(element1, occurrence1);
+        bag.addOccurrences(element2, occurrence2);
+        return bag.toImmutable();
+    }
+
+    @Override
+    public <T> ImmutableBag<T> withOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3)
+    {
+        MutableBag<T> bag = new HashBag<>(occurrence1 + occurrence2 + occurrence3);
+        bag.addOccurrences(element1, occurrence1);
+        bag.addOccurrences(element2, occurrence2);
+        bag.addOccurrences(element3, occurrence3);
+        return bag.toImmutable();
+    }
+
+    @Override
+    public <T> ImmutableBag<T> withOccurrences(T element1, int occurrence1, T element2, int occurrence2, T element3, int occurrence3, T element4, int occurrence4)
+    {
+        MutableBag<T> bag = new HashBag<>(occurrence1 + occurrence2 + occurrence3 + occurrence4);
```

Comment:
The pre-sizings are wrong. This map will have four elements, and could be pre-sized to 4, or not pre-sized. This implementation could easily lead to OutOfMemoryError and must be fixed.

My recommendation would be not to pre-size at all, nor call a constructor. You could replace this with a call to empty(), and then the implementation could be pulled up as a default method.

Reply:
Thanks for catching that. I will fix this soon.